app/services/liquid_assigns_service.rb

```diff
@@ -15,4 +18,45 @@ def assigns
 
   def assigns_pro
   end
+
+  # This method uses Liquid::VariableLookup to find all liquid variables from
+  # a given text. We use the list to know which project assign we need.
+  def assigns_from_content
+    return AVAILABLE_PROJECT_ASSIGNS if text.nil?
+
+    variable_lookup = Liquid::VariableLookup.parse(text)
+    return (variable_lookup.lookups & AVAILABLE_PROJECT_ASSIGNS)
+  end
+
+  def cached_drops(records)
+    return [] if records.empty?
+
+    type = records.first.class.to_s.underscore
+    cache_key = "liquid-project-#{project.id}-#{type.pluralize}:#{records.maximum(:updated_at).to_i}-#{records.count}"
+    drop_class = "#{type.camelize}Drop".constantize
+
+    Rails.cache.fetch(cache_key) do
+      records.map { |record| drop_class.new(record) }
+    end
+  end
+
+  def project_assigns
+    project_assigns = { 'project' => ProjectDrop.new(project) }
+
+    assigns_from_content.each do |var|
```

Comment:
```suggestion
    assigns_from_content.each do |assigns_type|
```
Or something more descriptive?